### PR TITLE
add Kernel.power/2

### PR DESCRIPTION
**Pull request by vans163**

### CODE REVIEW

**Comment by josevalim** [comment]
@vans163 can you please bump CI to v1.13? The operator is not available in previous versions.

File: nx/lib/nx/defn/kernel.ex [hunk 1]
```diff
@@ -411,6 +411,21 @@ defmodule Nx.Defn.Kernel do
...
+  It delegates to `Nx.power/2` (supports broadcasting).
+
+  ## Examples
+
+      defn power(a, b) do
+        a ** b
+      end
+
+  """
+  def left ** right when Kernel.and(is_number(left), is_number(right)), do: Kernel.**(left, right)
```
Comment (polvalente) [inline nx/lib/nx/defn/kernel.ex:426]:
```suggestion
  def unquote(:"**")(left, right) when Kernel.and(is_number(left), is_number(right)), do: Kernel.**(left, right)
```

Please change all occurences to something like this so it compiles on Elixir 1.12

Comment (josevalim) [reply nx/lib/nx/defn/kernel.ex:426]: I think we can bump to 1.13, especially because it will allow us to use the `~M` and `~V` sigils!

**Comment by polvalente** [comment]
> @vans163 can you please bump CI to v1.13? The operator is not available in previous versions.

@josevalim are we dropping Elixir 1.12?

File: nx/lib/nx/defn/kernel.ex (hunk 1 above)
Comment (polvalente) [reply nx/lib/nx/defn/kernel.ex:426]: Sure, makes sense!

**Comment by vans163** [comment]
> @vans163 can you please bump CI to v1.13? The operator is not available in previous versions.

Where is the CI config?

**Comment by polvalente** [comment]
> > @vans163 can you please bump CI to v1.13? The operator is not available in previous versions.
> 
> Where is the CI config?

It's in the .github directory. We should probably bump the requirement in all mix.exs files as well

**Pull request merged**